src/paperless_tesseract/parsers.py

```diff
@@ -41,6 +41,15 @@ def get_settings(self) -> OcrConfig:
         """
         return OcrConfig()
 
+    def get_pages_count(self, document_path, mime_type):
+        pages_count = None
+        if mime_type == "application/pdf":
+            import pikepdf
+
+            pdf = pikepdf.open(document_path)
```

Comment:
In the event of an exception, this might not be cleaned up.  `pikepdf.Pdf.open` can function as a context manager to ensure resources are cleaned

There's probably a few other spots to benefit with this as well

Reply:
Like this ?

```
with pikepdf.open(document_path) as pdf:
    pages_count = len(pdf.pages)
```

Reply:
What may work, but I know `with pikepdf.Pdf.open` does

Reply:
> pikepdf.Pdf.open

I don't understand, without using the import ? And the `open` method is not available in pikepdf.Pdf